CMakeLists.txt

```diff
@@ -25,6 +25,7 @@ option(ENABLE_PYTHON_BINDINGS "Enable Python bindings" ON)
 option(ENABLE_NODE_BINDINGS "Build NodeJs bindings" ON)
 option(ENABLE_CCACHE "Speed up incremental rebuilds via ccache" ON)
 option(ENABLE_COVERAGE "Build with coverage instrumentalisation" OFF)
+option(ENABLE_COMPILER_WARNINGS "Build with compiler warnings as errors" OFF)
```

Comment:
why not default on? this would be a good way to annoy everyone into keeping the output warning free

Reply:
We discussed before starting this work that having it default on may be untenable due to the multitudes of platform and toolchains we aim to compile on.

A way out of that is to deem one toolchain _the one_ and require no warnings there while being more lenient elsewhere.